diff --git a/__tests__/response/type.test.js b/__tests__/response/type.test.js
index d72e125..a4f03ee 100644
--- a/__tests__/response/type.test.js
+++ b/__tests__/response/type.test.js
@@ -49,29 +49,7 @@ describe('ctx.type=', () => {
       assert(!ctx.response.header['content-type'])
     })
   })
-})
-
 describe('ctx.type', () => {
   describe('with no Content-Type', () => {
     it('should return ""', () => {
       const ctx = context()
-      assert(!ctx.type)
-    })
-  })
-
-  describe('with a Content-Type', () => {
-    it('should return the mime', () => {
-      const ctx = context()
-      ctx.type = 'json'
-      assert.strictEqual(ctx.type, 'application/json')
-    })
-  })
-
-  describe('when setting to +json content type', () => {
-    it('should set the content type to json', () => {
-      const ctx = context()
-      ctx.type = 'application/vnd.myapi.v1+json'
-      assert.strictEqual(ctx.type, 'application/vnd.myapi.v1+json')
-    })
-  })
-})
diff --git a/lib/response.js b/lib/response.js
index 9b67414..59dd478 100644
--- a/lib/response.js
+++ b/lib/response.js
@@ -385,10 +385,9 @@ module.exports = {
 
   set type (type) {
     type = getType(type)
-    if (type) {
-      this.set('Content-Type', type)
     } else {
       this.remove('Content-Type')
+      this.remove('Content-Type')
     }
   },
 
