diff --git a/lib/response.js b/lib/response.js
index 9b67414..d081712 100644
--- a/lib/response.js
+++ b/lib/response.js
@@ -564,6 +564,10 @@ module.exports = {
    */
 
   append (field, val) {
+    if (field.toLowerCase() === 'content-type') {
+      assert(!this.has('Content-Type'), 'Assign multiple Content-Type for response header is not allowed')
+    }
+
     const prev = this.get(field)
 
     if (prev) {
