diff --git a/__tests__/application/index.test.js b/__tests__/application/index.test.js
index dbb1a8b..5a4c399 100644
--- a/__tests__/application/index.test.js
+++ b/__tests__/application/index.test.js
@@ -73,11 +73,3 @@ describe('app', () => {
     assert.strictEqual(app.compose, compose)
   })
 
-  it('should have a static property exporting `HttpError` from http-errors library', () => {
-    const CreateError = require('http-errors')
-
-    assert.notEqual(Koa.HttpError, undefined)
-    assert.deepStrictEqual(Koa.HttpError, CreateError.HttpError)
-    assert.throws(() => { throw new CreateError(500, 'test error') }, Koa.HttpError)
-  })
-})
diff --git a/__tests__/application/response.test.js b/__tests__/application/response.test.js
index cd94865..bf73999 100644
--- a/__tests__/application/response.test.js
+++ b/__tests__/application/response.test.js
@@ -3,7 +3,6 @@
 const { describe, it } = require('node:test')
 const request = require('supertest')
 const assert = require('node:assert/strict')
-const Koa = require('../..')
 
 describe('app.response', () => {
   const app1 = new Koa()
@@ -110,5 +109,3 @@ describe('app.response', () => {
     return request(app8.callback())
       .get('/')
       .expect(200)
-  })
-})
diff --git a/__tests__/response/type.test.js b/__tests__/response/type.test.js
index d72e125..a921aaa 100644
--- a/__tests__/response/type.test.js
+++ b/__tests__/response/type.test.js
@@ -3,7 +3,6 @@
 const { describe, it } = require('node:test')
 const context = require('../../test-helpers/context')
 const assert = require('node:assert/strict')
-
 describe('ctx.type=', () => {
   describe('with a mime', () => {
     it('should set the Content-Type', () => {
@@ -48,30 +47,3 @@ describe('ctx.type=', () => {
       assert(!ctx.type)
       assert(!ctx.response.header['content-type'])
     })
-  })
-})
-
-describe('ctx.type', () => {
-  describe('with no Content-Type', () => {
-    it('should return ""', () => {
-      const ctx = context()
-      assert(!ctx.type)
-    })
-  })
-
-  describe('with a Content-Type', () => {
-    it('should return the mime', () => {
-      const ctx = context()
-      ctx.type = 'json'
-      assert.strictEqual(ctx.type, 'application/json')
-    })
-  })
-
-  describe('when setting to +json content type', () => {
-    it('should set the content type to json', () => {
-      const ctx = context()
-      ctx.type = 'application/vnd.myapi.v1+json'
-      assert.strictEqual(ctx.type, 'application/vnd.myapi.v1+json')
-    })
-  })
-})
diff --git a/lib/application.js b/lib/application.js
index 5a0a972..607eb3d 100644
--- a/lib/application.js
+++ b/lib/application.js
@@ -15,7 +15,6 @@ const onFinished = require('on-finished')
 const compose = require('koa-compose')
 const statuses = require('statuses')
 const { HttpError } = require('http-errors')
-
 const request = require('./request')
 const response = require('./response')
 const context = require('./context')
@@ -23,8 +22,6 @@ const isStream = require('./is-stream.js')
 const only = require('./only.js')
 
 /** @typedef {typeof import ('./context') & {
- *  app: Application
- *  req: import('http').IncomingMessage
  *  res: import('http').ServerResponse
  *  request: KoaRequest
  *  response: KoaResponse
@@ -51,22 +48,6 @@ function getAsyncLocalStorage (options) {
 module.exports = class Application extends Emitter {
   /**
    * Initialize a new `Application`.
-   *
-   * @api public
-   */
-
-  /**
-   *
-   * @param {object} [options] Application options
-   * @param {string} [options.env='development'] Environment
-   * @param {string[]} [options.keys] Signed cookie keys
-   * @param {boolean} [options.proxy] Trust proxy headers
-   * @param {number} [options.subdomainOffset] Subdomain offset
-   * @param {string} [options.proxyIpHeader] Proxy IP header, defaults to X-Forwarded-For
-   * @param {number} [options.maxIpsCount] Max IPs read from proxy IP header, default to 0 (means infinity)
-   * @param {function} [options.compose] Function to handle middleware composition
-   * @param {boolean} [options.asyncLocalStorage] Enable AsyncLocalStorage, default to false
-   *
    */
 
   constructor (options) {
@@ -91,7 +72,6 @@ module.exports = class Application extends Emitter {
     if (options.asyncLocalStorage) {
       if (v8.startupSnapshot?.isBuildingSnapshot?.()) {
         this.ctxStorage = null
-        v8.startupSnapshot.addDeserializeCallback(({ app, options }) => {
           app.ctxStorage = getAsyncLocalStorage(options)
         }, { app: this, options })
       } else {
@@ -103,13 +83,6 @@ module.exports = class Application extends Emitter {
   /**
    * Shorthand for:
    *
-   *    http.createServer(app.callback()).listen(...)
-   *
-   * @param {Mixed} ...
-   * @return {import('http').Server}
-   * @api public
-   */
-
   listen (...args) {
     debug('listen')
     const server = http.createServer(this.callback())
